conftest.py

```diff
@@ -97,7 +97,11 @@ def device(request, device_params):
 
     num_devices = ttl.device.GetNumPCIeDevices()
     assert device_id < num_devices, "CreateDevice not supported for non-mmio device"
-    device = ttl.device.CreateDevice(device_id=device_id, **device_params)
+    # TODO: 11059 move dispatch_core_type to device_params when all tests are updated to not use WH_ARCH_YAML env flag
+    dispatch_core_type = ttl.device.CoreType.WORKER
+    if ("WH_ARCH_YAML" in os.environ) and os.environ["WH_ARCH_YAML"] == "wormhole_b0_80_arch_eth_dispatch.yaml":
```

Comment:
Hmm... what are the plans to get rid of its usage within test files and test scripts if we're still using it?

Is the expectation that users will have specify `CoreType`?

Reply:
can we rename this to `ETH_DISPATCH_ARCH_YAML` just so it isn't WH specific?

Reply:
WH_ARCH_YAML should be deleted from the codebase, so I don't think renaming it is necessary. I was expecting either a `eth_dispatch_device_fixture` where `ttl.device.CreateDevice(device_id=device_id, dispatch_core_type=ttl.device.CoreType.ETH, **device_params)` is hardcoded, or a generic one but the dispatch_core_type is coming from `device_params`

Reply:
Is there a way to further restrict its usage?

This might be a bit draconian, but perhaps here we `unset` `WH_ARCH_YAML` before passing the dispatch core type, and at the runtime level we error out if we see `WH_ARCH_YAML` set? Just a thought.

This would only apply to pytests, meaning using `WH_ARCH_YAML` w/ gtests is prohibited.

Reply:
Hmm interesting. I think the place where we would have to check for `WH_ARCH_YAML` would have to be after device creation, since a lot of tests are setting the env after. I'm happy with announcing that this env flag is no longer doing anything in runtime and removing it from codebase soon.

Reply:
Aren't the core descriptors ingested during device creation? How does setting `WH_ARCH_YAML` after device creation on current main do anything? I thought it must be set before

Reply:
No, the WH_ARCH_YAML is currently parsed every time any API tries to get dispatch core data. I think there are tests where this is true. Not sure if it's buggy or not